### PR TITLE
update gem to have full api coverage per documentation spec

### PR DESCRIPTION
**Pull request by dustinsgoodman**
AdRoll released a v1 of their [API documentation](https://developers.adroll.com/crud-api/reference.html). I wanted to get the gem to spec with the latest from AdRoll and get some in gem documentation.

### Changes
* Added specs for all specs
* Added full support for Uhura
* Removed unfulfilled services and undocumented services that couldn't be confirmed
* Added endpoints to undocumented APIs
* Conform project to ruby styleguide best practices

### CODE REVIEW

File: lib/adroll/api/ad.rb [hunk 1]
```diff
@@ -3,38 +3,42 @@
...
       def set_outline(params)
-        call_api(:get, __method__, validate_params(params))
+        call_api(:get, __method__, sanitize_params(params))
       end
 
       private
 
-      def validate_params(params)
-        params.reject { |key, value| !WHITELIST_PARAMS.include?(key) || value.nil? }
+      def sanitize_params(params)
```
Comment (ndavis) [inline lib/adroll/api/ad.rb]: The sanitize_params method is the same in all the services so let's move it to the AdRoll::Api::Service and AdRoll::Uhura::Service

File: lib/adroller/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Adroller
-  VERSION = '1.6.3'
+  VERSION = '1.6.3'.freeze
```
Comment (ndavis) [inline lib/adroller/version.rb]: You haven't updated the version number  or would you rather do that in master?

File: lib/adroll/api/ad.rb (hunk 1 above)
Comment (dustinsgoodman) [inline lib/adroll/api/ad.rb]: Not quite. Although it performs the same, the WHITELIST_PARAMS is specific to the file in which the method belongs.

File: lib/adroller/version.rb (hunk 2 above)
Comment (dustinsgoodman) [inline lib/adroller/version.rb]: master. The release isn't valid until this is merged. 😄

File: lib/adroll/api/ad.rb (hunk 1 above)
Comment (ndavis) [inline lib/adroll/api/ad.rb]: Change is tabled for later.

**Pull request merged**